kmp-nativecoroutines-core/src/nativeCoroutinesMain/kotlin/com/rickclephas/kmp/nativecoroutines/NativeFlow.kt

```diff
@@ -23,10 +29,16 @@ typealias NativeFlow<T> = (onItem: NativeCallback<T>, onComplete: NativeCallback
  */
 fun <T> Flow<T>.asNativeFlow(scope: CoroutineScope? = null): NativeFlow<T> {
     val coroutineScope = scope ?: defaultCoroutineScope
-    return (collect@{ onItem: NativeCallback<T>, onComplete: NativeCallback<NativeError?> ->
+    return (collect@{ onItem: NativeCallback2<T, () -> Unit>,
+                      onComplete: NativeCallback<NativeError?>,
+                      onCancelled: NativeCallback<NativeError> ->
         val job = coroutineScope.launch {
             try {
-                collect { onItem(it) }
+                collect {
+                    suspendCoroutine<Unit> { cont ->
+                        onItem(it) { cont.resume(Unit) }
```

Comment:
This will freeze the continuation resulting in a `InvalidMutabilityException`: https://github.com/Kotlin/kotlinx.coroutines/issues/2363